docs/arch.md

```diff
@@ -87,23 +87,17 @@ There are these main tasks scheduled:
 ### Scheduling of Gatherer
 Gatherer is using this logic to start information gathering from the cluster, and it is handled in [periodic.go](pkg/controller/periodic/periodic.go).
 
-The periodic is using a producer/consumer queue. It is periodically adding the gatherer to queue. The queue has a limit for maximally one gatherer of the name in queue and because we only have gatherer "config" it is only this one in queue.
-The adding to queue is run in periodic.periodicTrigger, which is started from periodic.Run function. 
-Function periodicTrigger tries to add a new one (blocking if the queue is full) every interval/2.
+So far we have only 1 Gatherer(called `clusterconfig`), it has several gather-functions each collecting different data from the cluster.
+The workflow of the gather-functions is managed by the Gatherer and NOT `periodic.go`.
+1 Gatherer runs at one time, this is because we only have 1 Gatherer at the moment and its doubtful that we will have more anytime soon. (ie.: we can and concurrency here when its needed)
```

Comment:
What about `Only one gatherer runs at a time.....` and I think we should avoid any speculation here. :)